routes/actions.js

```diff
@@ -347,4 +347,12 @@ function sendRegistrationForm(req, res, formInfo) {
   });
 }
 
+// checks for external redirect in returnTo, ignores if present and redirects
+function returnToPath(req, res) {
+    let returnTo = req.query.returnTo;
+    if (returnTo.includes('http')) {
```

Comment:
This is not a good way to check for external URLs. First, there are many protocols -- http, https, ftp, mailto, etc. -- all of which we have to prevent redirects to. Second, checking anywhere within the string produces false positives such as "the-http-handbook". 

With your code, try http://localhost/signin?returnTo=//oops.com

That's what's known as a protocol-relative URL. The string http or https is not present in the URL; on lib.reviews it would redirect to https since lib.reviews itself is on https.

The approach I'd suggest is a regular expression that looks for a / at the beginning, followed by any character that is _not_ a slash. That seems like a reliable way to ensure that we're redirecting to a path on the same way.

The way to do this is something along these lines:

```javascript
const localPathRegex = new RegExp('^\/[^/]');
if (!localPathRegex.test(returnTo)) // Suppress redirect, could be a non-local resource
```

If you're unfamiliar with regular expressions,  you can use https://regex101.com/r/OxaMCS/1 to get an explanation of what this check does, and to perform your own sanity checks.

Reply:
wouldn't changing the search string from http to // do the same thing?

Is there an advantage to using a regexp for this?